### PR TITLE
Pigeonhole man

### PR DESCRIPTION
**Pull request by cmouse**
JIRA: DOV-5632

### CODE REVIEW

**Review by sirainen — DISMISSED** [review]

File: source/pigeonhole-man/option-d.inc.in [hunk 1]
```diff
@@ -0,0 +1,9 @@
+**-d**
+   Don't write the binary to *out-file*, but write a textual dump of the
+   binary instead. In this context, the *out-file* value '-' has special
+   meaning: it causes the the textual dump to be written to **stdout**.
+   The *out-file* argument may also be omitted, which has the same
+   effect as '-'. The output is identical to what the
+   :man:`sieve-dump(1)` command produces for a compiled Sieve binary
+   file. Note that this option is not allowed when the *out-file*
+   argument is a directory.
```
Comment (stephanbosch) [inline source/pigeonhole-man/option-d.inc.in:9]: The -d argument is different between sievec and other sieve tools. This file suggests those were merged, which isn't possible.

File: source/pigeonhole-man/option-x-ext.inc.in [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+   listed will be enabled. Unknown extensions are ignored and a warning
+   is produced.
+
+   For example **-x** "+editheader -enotify" will enable the editheader
+   extension and disable the enotify extension. The rest of the active
+   extensions depends on the :pigeonhole:ref:`sieve_extensions` and
+   :pigeonhole:ref:`sieve_global_extensions` settings. By default, i.e. when
+   :pigeonhole:ref:`sieve_extensions` and :pigeonhole:ref:`sieve_global_extensions` remain unconfigured,
+   all supported extensions are available, except for deprecated extensions
+   or those that are still under development.
```
Comment (stephanbosch) [inline source/pigeonhole-man/option-x-ext.inc.in:16]: Actually, not true anymore. Also extensions that have security implications ("editheader") and those that need explicit configuration ("spamtest" and friends) are not enabled by default.

File: source/pigeonhole-man/sieve-filter.1.rst.in [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+**-C**
+   Force compilation. By default, the compiled binary is stored on disk.
+   When this binary is found during the next execution of
+   **sieve-filter** and its modification time is more recent than the
+   script file, it is used and the script is not compiled again. This
+   option forces the script to be compiled, thus ignoring any present
+   binary. Refer to **sievec**\ (1) for more information about Sieve
+   compilation.
+
```
Comment (stephanbosch) [inline source/pigeonhole-man/sieve-filter.1.rst.in:92]: Sieve-filter should also accept -d.

File: source/pigeonhole-man/option-x-ext.inc.in (hunk 2 above)
Comment (cmouse) [reply source/pigeonhole-man/option-x-ext.inc.in:16]: can you be more explicit about what is not true anymore?

File: source/pigeonhole-man/option-d.inc.in (hunk 1 above)
Comment (stephanbosch) [reply source/pigeonhole-man/option-d.inc.in:9]: For most tools -d adds its own <dumpfile> parameter. For sievec, the optional second <outfile> argument is affected and -d does not add its own parameter.

File: source/pigeonhole-man/option-x-ext.inc.in (hunk 2 above)
Comment (stephanbosch) [reply source/pigeonhole-man/option-x-ext.inc.in:16]: "all supported extensions are available, except for deprecated extensions  or those that are still under development"

So, the exceptions now also include the security-sensitive extensions and those that require explicit configuration.

File: source/pigeonhole-man/option-x-ext.inc.in [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+   listed will be enabled. Unknown extensions are ignored and a warning
+   is produced.
+
+   For example **-x** "+editheader -enotify" will enable the editheader
+   extension and disable the enotify extension. The rest of the active
+   extensions depends on the :pigeonhole:ref:`sieve_extensions` and
+   :pigeonhole:ref:`sieve_global_extensions` settings. By default, i.e. when
+   :pigeonhole:ref:`sieve_extensions` and :pigeonhole:ref:`sieve_global_extensions` remain unconfigured,
+   all supported extensions are available, except for deprecated extensions, security sensitive extension,
+   ones that require explicit configuration or those that are still under development.
```
Comment (stephanbosch) [inline source/pigeonhole-man/option-x-ext.inc.in:16]: Several problems in this sentence; try this:

", except for deprecated extensions, security-sensitive extensions, extensions that require explicit configuration, and those that are still under development."

File: source/pigeonhole-man/pigeonhole.7.rst.in [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+-  The LDA Sieve plugin for Dovecot's Local Delivery Agent (LDA)
+   (:man:`dovecot-lda(1)`) that facilitates the actual Sieve filtering
+   upon delivery.
+
+-  The ManageSieve service that implements the ManageSieve protocol
+   through which users can remotely manage Sieve scripts on the server.
+
+-  A plugin for Dovecot's :man:`doveadm(1)` command line tool that adds
+   new :man:`doveadm-sieve(1)` commands for management of Sieve
+   filtering.
```
Comment (stephanbosch) [inline source/pigeonhole-man/pigeonhole.7.rst.in:57]: The IMAPSieve plugin and FILTER=SIEVE plugin should probably also be mentioned here now.

File: source/pigeonhole-man/pigeonhole.7.rst.in [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+The functionality and configuration of the LDA Sieve plugin and the
+ManageSieve service is described in detail in the README and INSTALL
+files contained in the Pigeonhole package and in the Dovecot Wiki
+<http://wiki2.dovecot.org/Pigeonhole>.
```
Comment (stephanbosch) [inline source/pigeonhole-man/pigeonhole.7.rst.in:62]: Outdated URL.

File: source/pigeonhole-man/pigeonhole.7.rst.in [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+.. include:: reporting-bugs.inc
+
+AUTHOR
+======
+
+Pigeonhole <http://pigeonhole.dovecot.org> and its manual pages were
+written by the Pigeonhole authors <http://pigeonhole.dovecot.org/doc/AUTHORS>,
+mainly Stephan Bosch <stephan at rename-it.nl>, and are licensed under the terms of the
+LGPLv2.1 license, which is the same license as Dovecot, see
+<http://dovecot.org/doc/COPYING> for details.
```
Comment (stephanbosch) [inline source/pigeonhole-man/pigeonhole.7.rst.in:91]: Should probably unify this with Dovecot core.

File: source/pigeonhole-man/sieve-dump.1.rst.in [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+SYNOPSIS
+========
+
+**sieve-dump** [*options*] *sieve-binary* [*out-file*]
+
+DESCRIPTION
+===========
+
+The **sieve-dump** command is part of the Pigeonhole Project (:man:`pigeonhole(7)`),
+which adds Sieve (`RFC 5228 <http://tools.ietf.org/html/rfc5228/>`_) and
```
Comment (stephanbosch) [inline source/pigeonhole-man/sieve-dump.1.rst.in:21]: IETF tool URLs are deprecated and now redirect to rfc-editor. Should probably update these.

File: source/pigeonhole-man/sieve-filter.1.rst.in [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+.. include:: option-d.inc
```
Comment (stephanbosch) [inline source/pigeonhole-man/sieve-filter.1.rst.in:93]: The description for -d included here is for sievec and not for the -d <dumpfile> structure used by other commands like sieve-filter. This problem is therefore also present for sieve-test.

File: source/pigeonhole-man/sieve-dump.1.rst.in (hunk 8 above)
Comment (cmouse) [reply source/pigeonhole-man/sieve-dump.1.rst.in:21]: see #614

File: source/pigeonhole-man/pigeonhole.7.rst.in [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+NAME
+====
+
+pigeonhole - Overview of the Pigeonhole project's :ref:`Sieve <sieve>` support for the
+Dovecot secure :ref:`IMAP <imap>` and :ref:`POP3 <pop3>` server.
+
+DESCRIPTION
+===========
+
+Pigeonhole Project (:man:`pigeonhole(7)`) adds support for the
```
Comment (stephanbosch) [inline source/pigeonhole-man/pigeonhole.7.rst.in:16]: oh, this should be 'The Pigeonhole Project'

File: source/pigeonhole-man/pigeonhole.7.rst.in [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+-  A plugin for Dovecot's :man:`doveadm(1)` command line tool that adds
+   new :man:`doveadm-sieve(1)` commands for management of Sieve
+   filtering.
+
+-  :ref:`IMAPSieve plugin <pigeonhole_plugin_imapsieve>`, which provides
+   ability to attach Sieve scripts that are run for IMAP events in their mailboxes.
+
+-  :ref:`FILTER=SIEVE <pigeonhole_plugin_filter_sieve>` IMAP capability that
+   allows refiltering mails in a mailbox using Sieve scripts.
```
Comment (stephanbosch) [inline source/pigeonhole-man/pigeonhole.7.rst.in:62]: The other items begin with The or A

**Review by stephanbosch — APPROVED** [review]

**Pull request merged**